modules/programs/dircolors.nix

```diff
@@ -0,0 +1,221 @@
+      ".alz" = mkDefault "01;31";
+      ".ace" = mkDefault "01;31";
+      ".zoo" = mkDefault "01;31";
+      ".cpio" = mkDefault "01;31";
+      ".7z" = mkDefault "01;31";
+      ".rz" = mkDefault "01;31";
+      ".cab" = mkDefault "01;31";
+      ".wim" = mkDefault "01;31";
+      ".swm" = mkDefault "01;31";
+      ".dwm" = mkDefault "01;31";
+      ".esd" = mkDefault "01;31";
+      ".jpg" = mkDefault "01;35";
+      ".jpeg" = mkDefault "01;35";
+      ".mjpg" = mkDefault "01;35";
+      ".mjpeg" = mkDefault "01;35";
+      ".gif" = mkDefault "01;35";
+      ".bmp" = mkDefault "01;35";
+      ".pbm" = mkDefault "01;35";
+      ".pgm" = mkDefault "01;35";
+      ".ppm" = mkDefault "01;35";
+      ".tga" = mkDefault "01;35";
+      ".xbm" = mkDefault "01;35";
+      ".xpm" = mkDefault "01;35";
+      ".tif" = mkDefault "01;35";
+      ".tiff" = mkDefault "01;35";
+      ".png" = mkDefault "01;35";
+      ".svg" = mkDefault "01;35";
+      ".svgz" = mkDefault "01;35";
+      ".mng" = mkDefault "01;35";
+      ".pcx" = mkDefault "01;35";
+      ".mov" = mkDefault "01;35";
+      ".mpg" = mkDefault "01;35";
+      ".mpeg" = mkDefault "01;35";
+      ".m2v" = mkDefault "01;35";
+      ".mkv" = mkDefault "01;35";
+      ".webm" = mkDefault "01;35";
+      ".ogm" = mkDefault "01;35";
+      ".mp4" = mkDefault "01;35";
+      ".m4v" = mkDefault "01;35";
+      ".mp4v" = mkDefault "01;35";
+      ".vob" = mkDefault "01;35";
+      ".qt" = mkDefault "01;35";
+      ".nuv" = mkDefault "01;35";
+      ".wmv" = mkDefault "01;35";
+      ".asf" = mkDefault "01;35";
+      ".rm" = mkDefault "01;35";
+      ".rmvb" = mkDefault "01;35";
+      ".flc" = mkDefault "01;35";
+      ".avi" = mkDefault "01;35";
+      ".fli" = mkDefault "01;35";
+      ".flv" = mkDefault "01;35";
+      ".gl" = mkDefault "01;35";
+      ".dl" = mkDefault "01;35";
+      ".xcf" = mkDefault "01;35";
+      ".xwd" = mkDefault "01;35";
+      ".yuv" = mkDefault "01;35";
+      ".cgm" = mkDefault "01;35";
+      ".emf" = mkDefault "01;35";
+      ".ogv" = mkDefault "01;35";
+      ".ogx" = mkDefault "01;35";
+      ".aac" = mkDefault "00;36";
+      ".au" = mkDefault "00;36";
+      ".flac" = mkDefault "00;36";
+      ".m4a" = mkDefault "00;36";
+      ".mid" = mkDefault "00;36";
+      ".midi" = mkDefault "00;36";
+      ".mka" = mkDefault "00;36";
+      ".mp3" = mkDefault "00;36";
+      ".mpc" = mkDefault "00;36";
+      ".ogg" = mkDefault "00;36";
+      ".ra" = mkDefault "00;36";
+      ".wav" = mkDefault "00;36";
+      ".oga" = mkDefault "00;36";
+      ".opus" = mkDefault "00;36";
+      ".spx" = mkDefault "00;36";
+      ".xspf" = mkDefault "00;36";
+    };
+
+    home.file.".dir_colors".text = concatStringsSep "\n" ([ ]
+      ++ optional (cfg.extraConfig != "") cfg.extraConfig
+      ++ mapAttrsToList formatLine cfg.settings);
+
+    programs.bash.initExtra = mkIf cfg.enableBashIntegration ''
+      eval $(${pkgs.coreutils}/bin/dircolors -b ~/.dir_colors)
```

Comment:
I'm slightly concerned that this could cause problems if the terminal does not support colors.

I notice that the `dircolors -p` output includes a bunch of lines starting with `TERM` that seem to check whether the terminal can handle colors. Perhaps you could add these to the dircolors file?

Reply:
I actually intentially removed the `TERM` lines because they aren't exhaustive, and if your terminal isn't on that list (like Alacritty), the module silently fails. I figured, if a user's terminal doesn't support colors, they won't enable the `dircolors` module.

If a user has multiple terminals, and some don't support colors, they can add `TERM` lines to `extraConfig`. I thought it would be better to fully support the most common usecase out of the box: you enable `dircolors` because you have a terminal that supports colors and want to change colors.